### PR TITLE
Empty quick filters

### LINKED ISSUE #16001: User search is not handling special chars
**Describe the bug**
Incorrect symbols search '+', '&'
Get error for symbols search '[', ']' 

**To Reproduce**
Enter '+', '&', '[', ']' into main search 
or use the user search https://sandbox.open-metadata.org/settings/members/users?user=%5B
image

**Expected behavior**
Correct symbols search '+', '&'
Fix error for symbols search '[', ']' 

**Version:**
 - OpenMetadata version: 1.2.5 and current, maybe in another versions too

### PR DESCRIPTION
**Pull request by karanh37**
### Describe your changes:

1. Add Empty Quick Filters in Explore
2. Minor fixes related to search encoding. Fixes #16001 

https://github.com/open-metadata/OpenMetadata/assets/33024356/c3384944-671f-4e0e-ae7c-3ea2c3921f71

I worked on ... because ...

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

**Comment by harshach** [comment]
@karanh37 what test coverage we have for this

**Comment by karanh37** [comment]
@harshach - we have added two tests.
First test - We are individually checking each Null options and verifying if the API request is correct.
Second test - We are checking for Multiple Owners. We select `No Owner` and `Aaron Johnson` and verify multiple owner scenario is working fine.

**Comment by harshach** [comment]
we need to add tests for each of those fields where we expecting None to work. Having only one field test is not good enough

**Comment by karanh37** [comment]
@harshach - added tests for all `Owner` `Tag` `Tier` and `Domain`
Both single select and multi select cases added for all of above.

File: openmetadata-ui/src/main/resources/ui/src/components/SearchDropdown/SearchDropdown.tsx [hunk 1]
```diff
@@ -65,14 +67,18 @@ const SearchDropdown: FC<SearchDropdownProps> = ({
...
 
   const [isDropDownOpen, setIsDropDownOpen] = useState<boolean>(false);
   const [searchText, setSearchText] = useState('');
-  const [selectedOptions, setSelectedOptions] =
-    useState<SearchDropdownOption[]>(selectedKeys);
+  const [selectedOptions, setSelectedOptions] = useState<
+    SearchDropdownOption[]
+  >([]);
+  const [nullOptionSelected, setNullOptionSelected] = useState<boolean>(false);
+  const nullLabelText = `No ${label}`;
```
Comment (aniketkatkar97) [inline openmetadata-ui/src/main/resources/ui/src/components/SearchDropdown/SearchDropdown.tsx:81]: Are we not using localization for the label here?

**Review by aniketkatkar97 — COMMENTED** [review]
Few comments

File: openmetadata-ui/src/main/resources/ui/src/components/SearchDropdown/SearchDropdown.tsx (hunk 1 above)
Comment (karanh37) [reply openmetadata-ui/src/main/resources/ui/src/components/SearchDropdown/SearchDropdown.tsx:81]: We are not doing localisation in Quick Filters title as well, so didn't do here.

File: openmetadata-ui/src/main/resources/ui/cypress/e2e/Flow/AdvancedSearchQuickFilters.spec.ts [hunk 2]
```diff
@@ -11,41 +11,94 @@
...
 const ownerName = 'Aaron Johnson';
 
+const domainDetails = {
+  name: `cypress-domain-${uuid()}`,
+  displayName: `Cypress Domain QfTest`,
+  description: 'Cypress domain description',
+  domainType: 'Aggregate',
+  experts: [],
+  style: {},
+};
```
[the remark this answers is not in the document]
Comment (karanh37) [reply openmetadata-ui/src/main/resources/ui/cypress/e2e/Flow/AdvancedSearchQuickFilters.spec.ts]: Fixed

File: openmetadata-ui/src/main/resources/ui/src/components/SearchDropdown/SearchDropdown.tsx (hunk 1 above)
Comment (karanh37) [reply openmetadata-ui/src/main/resources/ui/src/components/SearchDropdown/SearchDropdown.tsx]: Fixed

**Review by aniketkatkar97 — APPROVED** [review]

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**